src/sbuf/MemBlob.cc

```diff
@@ -46,6 +42,13 @@ MemBlobStats::dump(std::ostream &os) const
     return os;
 }
 
+static auto &
+GetStatsInstance()
+{
+    static auto stats = new MemBlobStats();
```

Comment:
For consistency sake and to improve const-correctness, please add `const` that NoNewGlobals sketch contains:

```suggestion
    static const auto stats = new MemBlobStats();
```

Reply:
But this is not const, it's a read-write reference. GetStats returns the const version. Are you sure?

Reply:
```C++
static const auto stats = new MemBlobStats();
```

> But this is not const, it's a read-write reference. Are you sure?

The `stats` object is constant. The value of that pointer (i.e. the address of the allocated MemBlobStats object) never changes. That is what NoNewGlobals pattern is declaring.

The `*stats` object (i.e., the allocated MemBlobStats object) is not constant. NoNewGlobals is not marking it as such.

If we wanted to mark the allocated object as constant (in addition to marking the pointer as constant), we would do something like this instead (but then the function would also need to return a const reference):

```C++
const auto &
Foo() {
    static const auto * const foo = new Type();
    return *foo;
}
```

I have designed NoNewGlobals pattern to correctly handle typical cases (like GetStatsInstance()). If you have questions about that pattern, I am happy to collaborate to find answers. Meanwhile, please use that exact pattern where the compiler allows you to use it (instead of _starting_ with other patterns and leaving them in if they happen to compile). Fixing the same mistakes over and over again is costly.

Moreover, please start with "const auto foo = ..." in all contexts that do not require a reference. In most cases, that is the right pattern. In most cases, the compiler will tell you when that pattern is not applicable. Adjust as needed to remove `const`, add reference, etc., but _start_ with that simplest (and usually correct) pattern.

Reply:
Thanks